plugin/keyring_vault/vault_credentials.cc

```diff
@@ -0,0 +1,18 @@
+#include "vault_credentials.h"
+
+namespace keyring
+{
+
+const Secure_string& get_credential(const Vault_credentials &credentials, const Secure_string &key)
+{
+  static Secure_string empty_value;
```

Comment:
As to 2. I think that it would be destroyed when only main thread is left. Anyways, I have read that in order to make sure that initalization is thread safe compilers some times lock a mutex around local static variable. I will move this variable outside the function.